cmd/syft/internal/test/integration/encode_decode_cycle_test.go

```diff
@@ -43,26 +43,27 @@ func TestEncodeDecodeEncodeCycleComparison(t *testing.T) {
 			},
 			json: true,
 		},
-		{
-			name: cyclonedxjson.ID.String(),
-			redactor: func(in []byte) []byte {
-				// unstable values
-				in = regexp.MustCompile(`"(timestamp|serialNumber|bom-ref|ref)":\s*"(\n|[^"])+"`).ReplaceAll(in, []byte(`"$1": "redacted"`))
-				in = regexp.MustCompile(`"(dependsOn)":\s*\[(?:\s|[^]])+]`).ReplaceAll(in, []byte(`"$1": []`))
-				return in
-			},
-			json: true,
-		},
-		{
-			name: cyclonedxxml.ID.String(),
-			redactor: func(in []byte) []byte {
-				// unstable values
-				in = regexp.MustCompile(`(serialNumber|bom-ref|ref)="[^"]+"`).ReplaceAll(in, []byte{})
-				in = regexp.MustCompile(`<timestamp>[^<]+</timestamp>`).ReplaceAll(in, []byte{})
-
-				return in
-			},
-		},
+		// TODO: ignoring the `ref` field though does create stable results to compare, but the SBOM is fundamentally gutted and not worth comparing (find a better redaction or compare method)
+		//{
+		//	name: cyclonedxjson.ID.String(),
+		//	redactor: func(in []byte) []byte {
+		//		// unstable values
+		//		in = regexp.MustCompile(`"(timestamp|serialNumber|bom-ref|ref)":\s*"(\n|[^"])+"`).ReplaceAll(in, []byte(`"$1": "redacted"`))
+		//		in = regexp.MustCompile(`"(dependsOn)":\s*\[(?:\s|[^]])+]`).ReplaceAll(in, []byte(`"$1": []`))
+		//		return in
+		//	},
+		//	json: true,
+		//},
+		//{
+		//	name: cyclonedxxml.ID.String(),
+		//	redactor: func(in []byte) []byte {
+		//		// unstable values
+		//		in = regexp.MustCompile(`(serialNumber|bom-ref|ref)="[^"]+"`).ReplaceAll(in, []byte{})
+		//		in = regexp.MustCompile(`<timestamp>[^<]+</timestamp>`).ReplaceAll(in, []byte{})
+		//
+		//		return in
+		//	},
+		//},
```

Comment:
this is worth calling out -- I noticed that when redacting the `ref` field the SBOM generated is ultimately wrong (several empty dependency fields created). This makes the test fundamentally wrong, and needs to be rethought for this format.

Why was this dealt with on this PR? Changing the image test fixture lead to test breakage, which is how I found the problem.